plugin/azure/setup.go

```diff
@@ -0,0 +1,139 @@
+package azure
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/dns/mgmt/dns"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/pkg/fall"
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/coredns/coredns/plugin/pkg/upstream"
+	"github.com/mholt/caddy"
+)
+
+var log = clog.NewWithPlugin("azure")
+
+func init() {
+	caddy.RegisterPlugin("azure", caddy.Plugin{
+		ServerType: "dns",
+		Action:     setup,
+	})
+}
+
+func setup(c *caddy.Controller) error {
+	envSettings, keys, fall, err := parseCorefile(c)
+	if err != nil {
+		return err
+	}
+	up := upstream.New()
+	ctx := context.Background()
+	dnsClient := dns.NewRecordSetsClient(envSettings.Values[auth.SubscriptionID])
+	dnsClient.Authorizer, err = envSettings.GetAuthorizer()
+	if err != nil {
+		return c.Errf("\nfailed to create azure plugin:\n\n %v\n", err)
```

Comment:
Is it necessary to prefix and suffix with `\n`? I think those `\n` could be removed.